ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useRunReport.ts

```diff
@@ -0,0 +1,64 @@
+import { useCallback, useState } from 'react';
+
+import { runReportRequest } from 'services/ReportsService';
+import { ReportNotificationMethod } from 'services/ReportsService.types';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+export type UseSaveReportProps = {
+    onCompleted: () => void;
+};
+
+type Result = {
+    isRunCompleted: boolean;
+    isRunning: boolean;
```

Comment:
I know this is already merged, but something to consider in the future: instead of two booleans, a status variable with different constants.

This allows for adding other mutually exclusive states later, if needed.

We _may_ never need it here, or in other cases, but it is a good pattern to adopt, so that cases that do need multiple states are done the same way as just two states. 

Here's an article by Kent C. Dodds that explains this better than I can,
https://kentcdodds.com/blog/stop-using-isloading-booleans